doc/develop/modules.rst

```diff
@@ -1021,6 +1021,36 @@ maps, each of which has the following entries:
 - ``doc-url``: A URL pointing to the location of the official documentation for
   this blob
 
+Package manager dependencies
+============================
+
+Zephyr modules can describe dependencies from external package managers,
+currently only ``pip`` is supported.
+
+A west extension command ``west packages <manager>`` is available to list
+package dependencies for Zephyr and the current modules.
+Run ``west help packages`` for more details.
+
+Python pip
+----------
+
+Calling ``west packages pip`` would list individual packages and requirement
+files. Passing ``--install`` would install these if there's an active virtual
+environment.
+
+The following example demonstrates a module with a single ``pytest``
```

Comment:
> So maybe the implementation details should be in some west-independent .py file that everyone can leverage and the west packages command be just a front-end to this shared code. 

That would be ideal, and i'm fully in support of such improvement.
Just like we have `scripts/list_boards.py` and the `west boards` as front-end.

> Could be done later I guess

Yes, doing things step-wise is always good.
Though one thing to consider is that Zephyr users / companies not using `west` but passes Zephyr modules directly to CMake most likely has their own development environment setup handling, and thus chances are that they most likely won't benefit from such solution anyway.

> On that topic, there was an attempt from @tejlmand at https://github.com/zephyrproject-rtos/zephyr/pull/80782#discussion_r1846429010 to clarify the distinction between Zephyr module and west project. I feel like that important distinction has not made its way to the documentation.

That's a great suggestion. I just wonder where this should be placed so that it's not way down on a page 942 where no one will see it.

Reply:
> So there are Zephyr users who don't use `west` but do use module.yml files... except for this new feature that they can't leverage because they don't use `west`?
> 
> So maybe the implementation details should be in some west-independent `.py` file that everyone can leverage and the `west packages` command be just a front-end to this shared code. Could be done later I guess; should not affect `west` users.

Another option could be to only support Python packages to be listed as module requirements using an in-module `requirements.txt` file pointed out by the module (and not add support individual Python packages listed in the module YAML). This would have the benefit of leaning on what Python users are used to - and allow for easy integration in CI/downstream images/etc. which do not use west.

Reply:
That would be my preferred solution. It's simpler and introduces less of a "package manager" but more of a "convenience command" for installing module Python dependencies. I'd be interested in hearing what others think.

Reply:
Ping @tejlmand @marc-hb

Reply:
> I'd be interested in hearing what others think.

I have no special preferences.
But the `requirement-files` only support do has the benefit of KISS :slightly_smiling_face: